src/us/kbase/workspace/test/database/provenance/CreditMetadataEntryTest.java

```diff
@@ -0,0 +1,256 @@
+import us.kbase.workspace.database.provenance.Title;
+import us.kbase.workspace.database.WorkspaceUser;
+
+public class CreditMetadataEntryTest {
+
+	private static final String INCORRECT = "incorrect ";
+	// field names
+	private static final String USER = "user";
+	private static final String SCHEMA_VERSION = "credit metadata schema version";
+	private static final String CREDIT_METADATA = "credit metadata";
+
+	// field values
+	private static final String VERSION_STRING = "1.2.3";
+	private static final String VERSION_STRING_UNTRIMMED = "\n\r\n1.2.3\t\t";
+
+	// test error strings
+	private static final String INCORRECT_USER = INCORRECT + USER;
+	private static final String INCORRECT_SCHEMA_VERSION = INCORRECT + SCHEMA_VERSION;
+	private static final String INCORRECT_CREDIT_METADATA = INCORRECT + CREDIT_METADATA;
+
+	// error messages
+	private static final String USER_NON_NULL = "user cannot be null";
+	private static final String SCHEMA_VERSION_NON_NULL =
+			"creditMetadataSchemaVersion cannot be null or whitespace only";
+	private static final String CREDIT_METADATA_NON_NULL = "creditMetadata cannot be null";
+	private static final String TIMESTAMP_NON_NULL = "timestamp cannot be null";
+
+	private static final List<Contributor> CONTRIBUTOR_LIST = Arrays.asList(
+			Contributor.getBuilder("person", "Mr Blobby")
+					.build(),
+			Contributor.getBuilder("organisation", "The A Team")
+					.build());
+
+	private static final List<Title> TITLE_LIST = Arrays.asList(
+			Title.getBuilder("\t\t\f\t\n\rA Series of Unfortunate Elephants\n\n\n   ")
+					.build());
+
+	private static final CreditMetadata CREDIT_METADATA_EXAMPLE = CreditMetadata
+			.getBuilder("some:boring_identifier",
+					ResourceType.DATASET,
+					CONTRIBUTOR_LIST,
+					TITLE_LIST)
+			.withVersion("1.0.1-fml")
+			.build();
+
+	private static final String[] INVALID_SEM_VER_STRINGS = {
+			"v1",
+			"foo",
+			"01.1.0",
+			"alpha.01",
+			"beta1..1",
+			"build.01",
+			".build.01" };
+
+	private static final String EXP_EXC = "expected exception";
+
+	@Test
+	public void equals() throws Exception {
+		EqualsVerifier.forClass(CreditMetadataEntry.class).usingGetClass().verify();
+	}
+
+	/**
+	 * Check the resource metadata has the expected fields
+	 *
+	 * @param CreditMetadataEntry
+	 *                instance to check
+	 * @param user
+	 *                workspace user
+	 * @param version
+	 *                creditMetadataSchemaVersion string
+	 * @param creditMetadata
+	 *                instance to check
+	 */
+	private void assertResourceMetadataFields(
+			final CreditMetadataEntry creditMetadataEntry,
+			final WorkspaceUser user,
+			final String version,
+			final CreditMetadata creditMetadata) {
+		assertThat(INCORRECT_USER, creditMetadataEntry.getUser(), is(user));
+		assertThat(INCORRECT_SCHEMA_VERSION,
+				creditMetadataEntry.getCreditMetadataSchemaVersion(),
+				is(version));
+		TestCommon.assertCloseToNow(creditMetadataEntry.getTimestamp());
```

Comment:
I would submit specific timestamps rather than just making current timestamps so the test is exact. There's an `inst` function in `TestCommon` to save some typing